### ISSUE TITLE
Got Nan in loss calculation

### ISSUE THREAD

**Issue by begeekmyfriend** [tianrengao/SqueezeWave#4]
![image](https://user-images.githubusercontent.com/6031938/74581003-2d0aa980-4fe5-11ea-8247-f6771b3ae489.png)
![image](https://user-images.githubusercontent.com/6031938/74581006-33992100-4fe5-11ea-9e39-a94028d1dac5.png)
I have got negative value and `NaN` in loss calculation and it could not recover to normal training. It seems there is overflow in the intermediate variable. Is there any problem in the following calculation?
```python
loss = torch.sum(z*z)/(2*self.sigma*self.sigma) - log_s_total - log_det_W_total
```

**Comment by begeekmyfriend** [comment]
By the way, `a256` is all right and `a128` is always bad

**Comment by BohanZhai** [comment]
Would you mind tell me the batch size in your model?

**Comment by begeekmyfriend** [comment]
By the way the evaluation under configuration of `a256` and `c256` is all right under my checkpoint with 1577 epochs training. I am using my own mandarin data corpus. Therefore I think it is the configuration leads to the training failure.
[eval_a256_c256.zip](https://github.com/tianrengao/SqueezeWave/files/4341160/eval_a256_c256.zip)
![image](https://user-images.githubusercontent.com/6031938/76814948-2e730000-6837-11ea-9f40-0320d17a1dd8.png)
![image](https://user-images.githubusercontent.com/6031938/76814969-4054a300-6837-11ea-90b9-d108189d7f14.png)
![image](https://user-images.githubusercontent.com/6031938/76815009-5cf0db00-6837-11ea-8502-3b12c5932ccb.png)
![image](https://user-images.githubusercontent.com/6031938/76815029-6e39e780-6837-11ea-9c97-22fd5c68ca17.png)

**Comment by BohanZhai** [comment]
In my opinion, the large batch size may cause NaN. Also, because the dataset is changed, so the learning rate may change too.

**Comment by begeekmyfriend** [comment]
Maybe you are right I will try this.  Thanks for your reply and your brillliant jobs. There seems better cost performance than WaveGlow.

**Comment by gongchenghhu** [comment]
@begeekmyfriend  I have also got negative loss value. Have you solved it? Could you tell me how to solve this problem.

**Comment by begeekmyfriend** [comment]
The loss is essentially negative since it is composed of log liklihoods.

**Comment by gongchenghhu** [comment]
@begeekmyfriend Thank you very much!  I trained on Chinese biaobei dataset on a single GPU.  I don't know how many epochs it takes to get a good result. Now I have trained 230 epoch about 23689 step. The result on 22000step is not very good.  Do you have any idea?
[23000step-a256c256-result.zip](https://github.com/tianrengao/SqueezeWave/files/4391109/23000step-a256c256-result.zip)
![image](https://user-images.githubusercontent.com/45996835/77714633-74d51580-7014-11ea-89d5-0da3c535f485.png)

**Comment by begeekmyfriend** [comment]
In my hunble opinion the evaluation is not as good as that of WaveRNN. The SqueezeWave is light weight model for fast inference adapting for edge devices. It really achieve real time synthesis as a vocoder but it costs you the loss of a bit fidelity. You can wait until 1~2K epochs for better evaluation.

**Comment by gongchenghhu** [comment]
@begeekmyfriend Thanks for your quickly reply，and I will wait until 1~2K epochs.

**Comment by wizardk** [comment]
@begeekmyfriend Hi, did you compare the speed of SqueezeWave and WaveRNN without batched on single CPU?

**Comment by begeekmyfriend** [comment]
SqueezeWave is much more fast but with less fidelity because of less training sampling points.

**Comment by wizardk** [comment]
The reason for the NAN should refer to these : [https://github.com/NVIDIA/waveglow/issues/95](url) and [https://github.com/NVIDIA/waveglow/issues/123](url)
